### PR TITLE
Equality Check Bug Fix

### PR DESCRIPTION
**Pull request by HM33-Stu**
Equality check bug fix
Related PR (update to UG): https://github.com/AY2324S2-CS2103T-W11-1/tp/pull/266

**These will be our identity fields to check for equality:**
Company Name
Contact Name, Email, Phone Number (POC fields)
Description
Role 
Location

**Rationale**
Company Name + POC fields are not unique enough on their own.
Description is in there because its a compulsory field and users might use a placeholder, so its to discourage that. 
Application status another compulsory field is removed (to resolve the bug).
Role is a good first step but you can have the same role in 2 locations, so location is added

Let me know if you think a field should be added or removed.

It resolves the related bug raised;
#228 
image

### CODE REVIEW

**Comment by LWS49** [comment]
additional: add in FAQ the criteria for comparing internships, and that it will be rejected if you key in the same internship

File: src/test/java/seedu/address/model/internship/InternshipTest.java [hunk 1]
```diff
@@ -57,10 +56,6 @@ public void isSameInternship() {
         editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withContactEmail("alicer@exampler.com").build();
         assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
 
-        // different application status, all other attributes same -> returns false
-        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withApplicationStatus("rejected").build();
-        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
-
         // different description, all other attributes same -> returns false
```
Comment (LWS49) [inline src/test/java/seedu/address/model/internship/InternshipTest.java:59]: Please add checks for new identity fields!
// different role, all other attributes same -> returns false
        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withRole("Data Analyst").build();
        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
        
        // different location, all other attributes same -> returns false
        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withLocation("overseas").build();
        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));

**Review by LWS49 — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/address/model/internship/InternshipTest.java [hunk 2]
```diff
@@ -57,14 +56,18 @@ public void isSameInternship() {
...
         assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
 
+        // different location, all other attributes same -> returns false
+        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withLocation("overseas").build();
+        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
+
+        // different role, all other attributes same -> returns false
+        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withRole("Data Analyst").build();
+        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
+
```
Comment (HM33-Stu) [inline src/test/java/seedu/address/model/internship/InternshipTest.java:70]: @LWS49 Location and Role test cases added for isSameInternship()

File: src/test/java/seedu/address/model/internship/InternshipTest.java [hunk 3]
```diff
@@ -96,8 +99,24 @@ public void equals() {
...
+        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withDescription("Data Analyst Intern").build();
+        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
+
+        // different location -> returns false
+        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withLocation("overseas").build();
+        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
+
+        // different role -> returns false
+        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withRole("Data Analyst").build();
+        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
```
Comment (HM33-Stu) [inline src/test/java/seedu/address/model/internship/InternshipTest.java:120]: Added in test cases for the remaining identity fields as well

**Comment by HM33-Stu** [comment]
**Testing For Equality**
I will use the test case by the tester for all variants

**Current State**
isSame -> Company Name, POC, desc, role, location
equals -> Company Name, POC, desc, role, location

image


**If I add back application status to isSame**
image

**Review by CaptainKeqing — APPROVED** [review]
lgtm

**Pull request merged**